src/main/java/org/intellij/sdk/codesync/utils/ProjectUtils.java

```diff
@@ -55,4 +65,26 @@ public static String getPluginVersion() {
         }
         return version;
     }
+
+    public static boolean acquireLock() {
+        File lockFile = new File(LOCK_FILE);
```

Comment:
Maybe create lock file in the setupSystemDirectories part? Only try to acquire the lock here.

Reply:
For already synced repo this file will need to be created some other place, that is why I placed it here as well.

Reply:
I see. Got it. We run this createSystemDirectories everytime (no condition) in VSCode/Atom so that it create the required files if any is missing.

Reply:
I have updated the code to create directories at startup